HIP/hip-859.md

```diff
@@ -0,0 +1,140 @@
+---
+hip: 859
+title: Add support for gasConsumed field in REST APIs related to contract results
+author: Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>, Stoyan Panayotov <stoyan.panayotov@limechain.tech>
+working-group: Steven Sheehy <steven.sheehy@swirldslabs.com>, Nana Essilfie-Conduah <nana@swirldslabs.com>, Danno Ferrin <danno.ferrin@swirldslabs.com>, David Bakin <david.bakin@swirldslabs.com>
+type: Standards Track
+category: Mirror
+needs-council-approval: Yes
+status: Draft
+created: 2024-01-17
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/860
+updated: 2024-01-22
+---
+
+## Abstract
+
+Currently, consensus nodes export a `gasUsed` field for each EVM impacting transaction and the mirror node persists and exposes this as `gas_used` on the contract result REST APIs. This field value is calculated using the gas refund policy applied in the consensus node.
+This means that the value in this field represents how much gas the user was actually charged, not the amount of gas that the EVM consumed during the execution. This value can greatly differ from the actual amount that was used by the EVM. 
+
+Let's say we use a 20% refund policy for the `gasLimit` that is sent by the user.
+
+That means if the EVM consumes for example 55,500 gas, but the user is passing 400,000 gas limit, they will be charged the `max(55,500, 320,000)`, so it would be 320,000. That is the value calculated after we return the 20% of the gasLimit back to the user, which is 80,000.
+The logic compares the bigger of the two values - the gas consumed by the EVM and the 80% of the `gasLimit` that the user has passed.
+
+In this way we lose track of the actual gas that was consumed by the EVM.
+
+This HIP will add a new field `gas_consumed` to the mirror node contract result REST APIs, which will contain the actually consumed gas by the EVM.
+
+## Motivation
+
+Adding a `gas_consumed` field to the contract result REST APIs would enrich user experience by providing more information about the transaction. 
+This would allow users to better understand the gas refund policy and the actual gas consumption of their transactions. This would also allow users to understand how heavy their executed transaction was for the EVM.
```

Comment:
Personally I would exchange the order of "actual gas consumption of their transactions" and "better understand the gas refund policy" because I believe that _most_ use of this will be by contract developers looking to optimize their contracts (point #1 in "user stories" below).

Reply:
Fixed